### PR TITLE
Update `unified-language-server`

### PR DESCRIPTION
**Pull request by remcohaszing**
### Initial checklist

*   [x] I read the support docs 
*   [x] I read the contributing guide 
*   [x] I agree to follow the code of conduct 
*   [x] I searched issues and couldn’t find anything (or linked relevant results below) 
*   [x] If applicable, I’ve added docs and tests

### Description of changes

This updates `unified-language-server` to version 3. See https://github.com/unifiedjs/unified-language-server/releases/tag/3.0.0 for details.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -33,7 +33,7 @@ consume this package.
 ## Install
 
 This package is [ESM only](https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c).
-In Node.js (version 12.20+, 14.14+, or 16.0+), install with [npm][]:
+In Node.js (version 14.14+, or 16.0+), install with [npm][]:
```
Comment (wooorm) [inline readme.md]:
```suggestion
In Node.js (version 14.14+, 16.0+, or 18.0+), install with [npm][]:
```

File: readme.md [hunk 2]
```diff
@@ -55,7 +55,7 @@ Practical examples are coming soon.
 
 Projects maintained by the unified collective are compatible with all maintained
 versions of Node.js.
-As of now, that is Node.js 12.20+, 14.14+, and 16.0+.
+As of now, that is Node.js 14.14+, and 16.0+.
```
Comment (wooorm) [inline readme.md]:
```suggestion
As of now, that is Node.js 14.14+, 16.0+, and 18.0+.
```

File: readme.md (hunk 1 above)
Comment (remcohaszing) [reply readme.md]: Oops, this should have been part of #8.

Anyway, do you really want to explicitly state Node.js 18.0+ is supported if it’s already specified Node.js 16.0+ is supported? I get the point for Node 12 and 14, as they explicitly required minimal minor versions, but now IMO it’s getting redundant to mention every major Node.js version if we don’t rely in such specific features introduced in minor versions. I’m pretty sure most packages will work with future versions of Node.js.

Comment (wooorm) [reply readme.md]: Particularly in the compat docs, mentioning them a) helps clarify that this is modern and tested, b) the consistency with every project listing the supported release lines is nice, c) it makes future edits easier/smaller/clearer, e.g., when it’s `18.5+` or so

It’s less important in the install section, maybe just `(version 14.14 or later)` there is okay. On the other hand, it’s fine duplicating some info: users might see install but not look at compat

**Review by wooorm — APPROVED** [review]

**Pull request merged**